### PR TITLE
ss-167: Add budget details page

### PR DESCRIPTION
**Pull request by Holovashcenko**
image
image

### CODE REVIEW

**Comment by Holovashcenko** [comment]
image

**Comment by Holovashcenko** [comment]
image

**Comment by Holovashcenko** [comment]
image

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]
For modal paddings check other modals
example: Add a wallet

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]
Talk with @anton-otroshchenko about category dropdown

File: frontend/src/bundles/common/components/header/header.tsx [hunk 1]
```diff
@@ -63,7 +63,8 @@ const Header: React.FC<Properties> = ({
                 {token ? (
                     <div className={classNames(styles.headerBody, styles.tabs)}>
                         {(pathname === AppRoute.DASHBOARD ||
-                            pathname === AppRoute.BUDGETS) && (
+                            pathname === AppRoute.BUDGETS ||
+                            pathname.match(/^\/budgets\/\d+$/)) && (
```
Comment (Evcat6) [inline frontend/src/bundles/common/components/header/header.tsx:67]: I recomend to set this regex to variable

File: frontend/src/bundles/common/pages/budgets/budget-details/components/edit-budget-modal/edit-budget-modal.tsx [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+            currency: budget.currency,
+            recurrence: budget.recurrence,
+            categories: budget.categories,
+            startDate: budget.startDate,
+        },
+    });
+
+    const handleUpdateBudgetSubmit = useCallback((formData: FormData): void => {
+        // eslint-disable-next-line no-console
+        console.log(formData);
```
Comment (Evcat6) [inline frontend/src/bundles/common/pages/budgets/budget-details/components/edit-budget-modal/edit-budget-modal.tsx:56]: Why we need this console.log?

File: frontend/src/bundles/common/pages/budgets/budget-details/helpers/calculate-budget-details.helper.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+            break;
+        }
+    }
+    const lastDate = new Date(endDate).toLocaleDateString('en-US', {
+        month: 'short',
+        day: '2-digit',
+        year: 'numeric',
+    });
+    const timeRemainingMs = endDate.getTime() - Date.now();
+    const daysRemaining = Math.ceil(timeRemainingMs / (1000 * 60 * 60 * 24));
```
Comment (Evcat6) [inline frontend/src/bundles/common/pages/budgets/budget-details/helpers/calculate-budget-details.helper.ts:54]: Magic numbers

**Review by Evcat6 — CHANGES_REQUESTED** [review]

File: frontend/src/bundles/common/pages/budgets/budget-details/components/edit-budget-modal/edit-budget-modal.tsx (hunk 2 above)
Comment (Holovashcenko) [reply frontend/src/bundles/common/pages/budgets/budget-details/components/edit-budget-modal/edit-budget-modal.tsx:56]: It resolved in [https://github.com/BinaryStudioAcademy/bsa-winter-2022-2023-smartspend/pull/301/files#diff-239043c4b2074e50009c2ae567f6825686d1a2f949877d17e0dad66f1354852f](url)

File: frontend/src/bundles/common/components/header/header.tsx (hunk 1 above)
Comment (Holovashcenko) [reply frontend/src/bundles/common/components/header/header.tsx]: resolved in [https://github.com/BinaryStudioAcademy/bsa-winter-2022-2023-smartspend/pull/286/commits/b20760f84841bd84bb032c5ba118188e58a721e5](url)

File: frontend/src/bundles/common/pages/budgets/budget-details/helpers/calculate-budget-details.helper.ts (hunk 3 above)
Comment (Holovashcenko) [reply frontend/src/bundles/common/pages/budgets/budget-details/helpers/calculate-budget-details.helper.ts]: resolved in [https://github.com/BinaryStudioAcademy/bsa-winter-2022-2023-smartspend/pull/286/commits/b20760f84841bd84bb032c5ba118188e58a721e5](url)

File: frontend/src/bundles/common/pages/budgets/budget-details/budget-details.tsx [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+    const [active, setActive] = useState(false);
+
+    const handleCancel = useCallback(() => {
+        setActive(false);
+    }, []);
+    const handleModal = useCallback(() => {
+        setActive(true);
+    }, []);
+
+    // const { id } = useLocation();
```
Comment (Evcat6) [inline frontend/src/bundles/common/pages/budgets/budget-details/budget-details.tsx:150]: Does we need this?

**Review by Evcat6 — CHANGES_REQUESTED** [review]

File: frontend/src/bundles/common/pages/budgets/budget-details/budget-details.tsx (hunk 4 above)
Comment (Holovashcenko) [reply frontend/src/bundles/common/pages/budgets/budget-details/budget-details.tsx:150]: resolve in [https://github.com/BinaryStudioAcademy/bsa-winter-2022-2023-smartspend/pull/301/files#diff-7d399f4e833c64a50550a276c56abb5164f4a3ced886f28bdf20af73a21249b2](url)
Mikhaylo merged my branch and works with it

**Review by nikita-remeslov — APPROVED** [review]

**Pull request merged**